packages/slate/src/interfaces/element.js

```diff
@@ -367,6 +447,34 @@ class ElementInterface {
     return closest || null
   }
 
+  /**
+   * Get closest parent of node that matches an `iterator`.
+   *
+   * @param {List} path
+   * @param {Function} iterator
+   * @return {Node|Null}
+   */
+
+  getClosestByPath(path, iterator) {
```

Comment:
Why the need for this method? This seems like another optimization that I'd rather have in a separate PR, since the original `getClosest` already works by path.

That way, instead of creating these extra methods we can just use the optimized versions when we drop key support which is coming soon.